docs/install.rst

```diff
@@ -111,6 +179,75 @@ When using conda, ffmpeg can be installed as follows::
     conda install -c conda-forge ffmpeg
 
 
+Conda environments
+------------------
+.. _PyCharm: https://www.jetbrains.com/pycharm/download/#section=windows
+.. _PyCharm documentation: https://www.jetbrains.com/help/pycharm/conda-support-creating-conda-virtual-environment.html
+.. _Jupyter: https://jupyter.org/
+.. _uninstall instructions: https://docs.anaconda.com/anaconda/install/uninstall/
+
+If you have installed Pylake according to the Installation Instructions for Anaconda, then you should now have a separate environment for your Pylake work.
+You may be wondering why we needed to create a new environment for Pylake, and what an environment is.
+
+When using Python, you will quickly find that several packages that you can install depend on each other.
+For example, when installing Pylake, you could see that the installation of Pylake required numerous other packages to be installed as well.
+Most Python packages are continuously updated by their authors. Sometimes, these authors decide that certain existing functionalities need to be changed.
+This means that not all of them will be completely backwards compatible.
+Therefore, it can be challenging to find a suitable collection of packages that fulfills the requirements of the package you are trying to install, while ensuring that all packages are still compatible.
+
+One pragmatic solution to this is to maintain separate Python environments for different projects.
+This means the solver doesn't have a large number of dependencies to resolve.
+Anaconda is one solution to this problem. With Anaconda, you can have multiple installations of Python (with all their installed modules) installed on your computer.
+These installations are referred to as environments.
+
+**Why do we install Pylake in a separate environment by default?**
+
+Conda sources the packages it uses from a channel, these are locations where conda and the anaconda navigator search for packages.
+The default one is called Anaconda, but Pylake is available on a channel named `conda-forge`_.
+Conda forge and Anaconda both have different versions of different packages.
+Some of these are not compatible with each other.
+This is why it is wise to install Pylake into its own environment, and only source packages from the channel `conda-forge`_ in that environment.
+This helps prevent difficulties when trying to come up with a plan to install a package you request.
+
+**How do I set up my other tools to use the correct environment?**
+
+For most programs, it is just a matter of pointing them to the correct environment.
+If you prefer using the anaconda navigator, you can activate the environment by selecting it from the drop down menu `Applications on`.
+By default, the selected environment is `base`.
+
```

Comment:
VS Code? 🥺 
For that one, the instructions would be: press Ctrl+Shift+P, and run the command "Python: Select Interpreter". It should automatically list all conda environments in the list that pops up.

Reply:
That's what I tried, but while that works right off the bat on Ubuntu, it doesn't seem to work on a clean VS Code install on Windows (I tried a clean windows image for this, to make sure it wasn't just my system). While I can see that it attempts to activate the environment in the powershell prompt that runs the python script, it seems that the environment is not actually correctly activated.

See this open issue here: https://github.com/numpy/numpy/issues/15183#issuecomment-603575874

We have two options. Either instruct them to always start VSCode from the Anaconda navigator (this works and activates the environment prior to starting VS Code). Instruct them to start VS Code from the anaconda prompt *after* activating the environment. Or instruct them to change `settings.json`, which will make it work no matter from where you start it. If it was my computer, the last would have my preference, but it is also the most complicated instruction. What do you think?

```rst
Go to the extensions tab (CTRL + SHIFT + X)

Enter Python in the search field.

Click on the Python plugin (by Microsoft) and install it.

Restart VSCode.

CTRL + Shift + P

Type Python and select the pylake environment from the list.

CTRL + SHIFT + P

Open settings

Add the following lines to settings.json:

    "terminal.integrated.shell.windows": "C:\\WINDOWS\\System32\\cmd.exe",
    "terminal.integrated.shellArgs.windows": ["/K", "x:\\xxx\\Anaconda3\\Scripts\\activate.bat x:\\xxx\\Anaconda3"],

In these lines, you have to replace `x:\\xxx` by the location of your Anaconda installation. You can find out where it is installed by starting an Anaconda prompt and typing `where activate.bat`.
```

Just for reference, this is the error you get when you don't:

```
PS D:\> conda activate test_pylake_latest
PS D:\>  & 'C:\Conda\envs\test_pylake_latest\python.exe' 'c:\Users\JoepVanlier\.vscode\extensions\ms-python.python-2021.1.502429796\pythonFiles\lib\python\debugpy\launcher' '63797' '--' 'd:\test.py'
Traceback (most recent call last):
  File "C:\Conda\envs\test_pylake_latest\lib\site-packages\numpy\core\__init__.py", line 22, in <module>  
    from . import multiarray
  File "C:\Conda\envs\test_pylake_latest\lib\site-packages\numpy\core\multiarray.py", line 12, in <module>
    from . import overrides
  File "C:\Conda\envs\test_pylake_latest\lib\site-packages\numpy\core\overrides.py", line 7, in <module>
    from numpy.core._multiarray_umath import (
ImportError: DLL load failed while importing _multiarray_umath: The specified module could not be found.   

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "d:\test.py", line 1, in <module>
    import lumicks.pylake as lk
  File "C:\Conda\envs\test_pylake_latest\lib\site-packages\lumicks\pylake\__init__.py", line 4, in <module>
    from .file import *
  File "C:\Conda\envs\test_pylake_latest\lib\site-packages\lumicks\pylake\file.py", line 1, in <module>    
    import h5py
  File "C:\Conda\envs\test_pylake_latest\lib\site-packages\h5py\__init__.py", line 25, in <module>  
  File "C:\Conda\envs\test_pylake_latest\lib\site-packages\numpy\__init__.py", line 145, in <module>
    from . import core
  File "C:\Conda\envs\test_pylake_latest\lib\site-packages\numpy\core\__init__.py", line 48, in <module>
    raise ImportError(msg)
ImportError:

IMPORTANT: PLEASE READ THIS FOR ADVICE ON HOW TO SOLVE THIS ISSUE!

Importing the numpy C-extensions failed. This error can happen for
many reasons, often due to issues with your setup or how NumPy was
installed.

We have compiled some common reasons and troubleshooting tips at:

    https://numpy.org/devdocs/user/troubleshooting-importerror.html

Please note and check the following:

  * The Python version is: Python3.9 from "C:\Conda\envs\test_pylake_latest\python.exe"
  * The NumPy version is: "1.20.0"

and make sure that they are the versions you expect.
Please carefully study the documentation linked above for further help.

Original error was: DLL load failed while importing _multiarray_umath: The specified module could not be found.

PS D:\>  d:; cd 'd:\\'; & 'C:\Conda\envs\test_pylake_latest\python.exe' 'c:\Users\JoepVanlier\.vscode\extensions\ms-python.python-2021.1.502429796\pythonFiles\lib\python\debugpy\launcher' '63882' '--' 'd:\test.py' 
Traceback (most recent call last):
  File "C:\Conda\envs\test_pylake_latest\lib\site-packages\numpy\core\__init__.py", line 22, in <module>
    from . import multiarray
  File "C:\Conda\envs\test_pylake_latest\lib\site-packages\numpy\core\multiarray.py", line 12, in <module>
    from . import overrides
  File "C:\Conda\envs\test_pylake_latest\lib\site-packages\numpy\core\overrides.py", line 7, in <module>
    from numpy.core._multiarray_umath import (
ImportError: DLL load failed while importing _multiarray_umath: The specified module could not be found.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "d:\test.py", line 1, in <module>
    import lumicks.pylake as lk
  File "C:\Conda\envs\test_pylake_latest\lib\site-packages\lumicks\pylake\__init__.py", line 4, in <module>
    from .file import *
  File "C:\Conda\envs\test_pylake_latest\lib\site-packages\lumicks\pylake\file.py", line 1, in <module>
    import h5py
  File "C:\Conda\envs\test_pylake_latest\lib\site-packages\h5py\__init__.py", line 25, in <module>
    from . import _errors
  File "h5py\_errors.pyx", line 1, in init h5py._errors
  File "C:\Conda\envs\test_pylake_latest\lib\site-packages\numpy\__init__.py", line 145, in <module>
    from . import core
  File "C:\Conda\envs\test_pylake_latest\lib\site-packages\numpy\core\__init__.py", line 48, in <module>
    raise ImportError(msg)
ImportError:

IMPORTANT: PLEASE READ THIS FOR ADVICE ON HOW TO SOLVE THIS ISSUE!

Importing the numpy C-extensions failed. This error can happen for
many reasons, often due to issues with your setup or how NumPy was
installed.

We have compiled some common reasons and troubleshooting tips at:

    https://numpy.org/devdocs/user/troubleshooting-importerror.html

Please note and check the following:

  * The Python version is: Python3.9 from "C:\Conda\envs\test_pylake_latest\python.exe"
  * The NumPy version is: "1.20.0"

and make sure that they are the versions you expect.
Please carefully study the documentation linked above for further help.

Original error was: DLL load failed while importing _multiarray_umath: The specified module could not be found.
```

Reply:
Oh boy, such a mess. I understand your hesitation to include those instructions.

Perhaps we could give just the short instructions based on starting VS Code from the Navigator, and link to [this VS Code doc page](https://code.visualstudio.com/docs/python/environments#_conda-environments) (under the heading "Conda environments") for troubleshooting? We could even throw in that numpy discussion link as well for further troubleshooting. For novices, starting from the Navigator is probably just fine; and more experiences users can then easily figure the rest out themselves using the links provided.

What do you think?

Reply:
Yeah, referencing those pages is better. How about the following:

```rst
*VS Code*

#. Download `VS Code`_ and install it following the default installation instructions.

#. Start it when the installer finishes.

#. Go to the extensions tab (CTRL + SHIFT + X)

#. Enter Python in the search field.

#. Click on the Python plugin (by Microsoft) and install it.

#. Restart VS Code.

#. Open the Command Palette (CTRL + SHIFT + P) and type "Python: Select Interpreter". Here you should choose the pylake environment.

#. Close VS Code.

VS Code should now appear in your Anaconda Navigator list.
Make sure that you selected the pylake environment when starting VS Code from the Anaconda Navigator.
It should now be possible to use pylake in VS Code.

Under Windows, you will need to start VS Code from the Anaconda Navigator for it to use the correct environment.
For more information on how to get it to run without Anaconda Navigator on Windows see the `VS Code Environment Instructions`_ and this `numpy discussion`_.
```